### ISSUE TITLE
Errors Running TestAll After Install on Windows 10 (http://localhost:5101 and root at C:\hsds_data)

### ISSUE THREAD

**Issue by ron-kuhn** [HDFGroup/hsds#210]
I have the following errors during testall.py after installing on Windows 10 and running locally:

**Unit Test**
-  dset_util_test - testGetSelectionPagination - 4.1GB selection had to be commented out

**Integ Test**
- domain_test - testGetDomainVerbose and testGetDomainsVerbose both failed num_groups, num_datasets, total_size, num_objects (comments states "restore when sqlite changes are complete"
- group_test - testGetByPath fails  because path comes back as /g1\\g1.1 instead of /g1/g1.1
- dataset_test - testGet fails because alias comes back as /g1\\g1.1\\dset1.1.1 instead of /g1/g1.1/dset1.1.1
- dataset_test - testGetVerbose fails because num_chunks and allocated_size are not returned

I either fixed or bypassed these errors to get through all the other tests. Are these expected for my environment? Next step for me is to configure for Minio.

**Comment by ron-kuhn** [comment]
I configured my HSDS (Windows 10) to use my local MinIO (WIndows 10). Now these are fixed:

- domain_test - testGetDomainVerbose and testGetDomainsVerbose both failed num_groups, num_datasets, total_size, num_objects (comments states "restore when sqlite changes are complete"
- dataset_test - testGetVerbose fails because num_chunks and allocated_size are not returned

But these are still bad:
- group_test - testGetByPath fails because path comes back as /g1\g1.1 instead of /g1/g1.1
- dataset_test - testGet fails because alias comes back as /g1\g1.1\dset1.1.1 instead of /g1/g1.1/dset1.1.1

**Comment by jreadey** [comment]
I found the cause of the first two - there's a problem with the windows-style paths that are returned as HSDS walks through the directory tree.  Should have a fix soon.  It's not a problem with Minio as it always returns unix style paths.

I think there's a similar root cause for the last two issues.  I'll look into it.

**Comment by jreadey** [comment]
@ron-kuhn - Can you try out the "unboundlocal" branch?   I put some updates there that I think should resolve all test failures on Windows either with or without Minio.

**Comment by ron-kuhn** [comment]
I sure will test your fixes. Can you tell me how to build the HDF5 Rest VOL library using windows, so that my HDF5-based library can be used to directly read from HSDS? There is a build_vol_cmake.bat file but it only contains comments.

**Comment by ron-kuhn** [comment]
With your latest changes, everything is working (including other changes I made to get to work like python cmd). I only tested the MinIO configuration. FYI... HSDS production environment will be linux-based but developer PC's are windows. Any help with HDF5 vol-rest build in Windows?

**Comment by jhendersonHDF** [comment]
Hi @ron-kuhn,

the last time that I built the REST VOL on Windows, I used the [CMake GUI](https://cmake.org/cmake/help/latest/manual/cmake-gui.1.html) to generate Visual Studio .sln solution files and then let VS build the connector. I'd like to eventually flesh out a .bat file for that, but hadn't gotten around to it yet. The connector could also probably be built with MinGW or similar on Windows, but I haven't tried that.
